src/handle.ml

```diff
@@ -77,6 +77,49 @@ let handle_tactic : sig_state -> Proof.t -> p_tactic -> Proof.t =
   | P_tac_print
   | P_tac_proofterm
   | P_tac_focus(_)      -> assert false (* Handled above. *)
+  | P_tac_assert(t,h)     ->
```

Comment:
Here is an alternative implementation for the tactic:
```ocaml
       (* Build the proof term [(λ (h : t), _) _]. *)
       let h = match h with None -> Pos.none "H" | Some(h) -> h in
       let t = Pos.none (P_Appl(Pos.none (PAbst([(h,t)], Pos.none P_Wild)), Pos.none P_Wild)) in
       (* Scoping the term in the goal's environment. *)
       let env = fst (Proof.Goal.get_type g) in
       let t = fst (Scope.scope_term StrMap.empty ss env t) in
       (* Refine using the built term. *)
       handle_refine t
```
Note that it does not make use of any low-level Bindlib stuff. There is still one thing wrong here: there could well be a variable called ["H"] in the environment. This should be checked, and a "fresh" variable name should be generated (say, the first "Hn", with "n" a natural number, such that "Hn" is not in the environment).